wgpu-hal/src/metal/command.rs

```diff
@@ -1295,5 +1295,6 @@ impl Drop for super::CommandEncoder {
         unsafe {
             self.discard_encoding();
         }
+        self.counter.sub(1);
```

Comment:
Again, I may be missing something, but I think that, since `self.counter` is an entirely separate counter from `wgpu_hal::metal::device::counters.command_encoders`, this `sub` has no effect, since it only affects the counter that's about to be destroyed.

Reply:
I think this applies to the counters in the other backends as well.

Reply:
I wrapped `HalCounters` in an `Arc`.